### PR TITLE
New fields: numerical var and varm

### PR DESCRIPTION
**Pull request by vitkl**
These fields are needed when the dataset is split into batches of both cells and variables.

### CODE REVIEW

File: scvi/data/fields/_var_field.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+from ._base_field import BaseAnnDataField
+
+logger = logging.getLogger(__name__)
+
+
+class BaseVarField(BaseAnnDataField):
+    """
+    An abstract AnnDataField for .var attributes in the AnnData data structure.
+    Parameters
```
Comment (martinkim0) [inline scvi/data/fields/_var_field.py:17]:
```suggestion
    An abstract AnnDataField for .var attributes in the AnnData data structure.
    
    Parameters
```

File: scvi/data/fields/_var_field.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    @property
+    def is_empty(self) -> bool:  # noqa: D102
+        return self._is_empty
+
+
+class NumericalVarField(BaseVarField):
+    """
+    An AnnDataField for numerical .var attributes in the AnnData data structure.
+    Parameters
```
Comment (martinkim0) [inline scvi/data/fields/_var_field.py:61]:
```suggestion
    An AnnDataField for numerical .var attributes in the AnnData data structure.
    
    Parameters
```

File: scvi/data/fields/_varm_field.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    def attr_name(self) -> str:  # noqa: D102
+        return self._attr_name
+
+
+class VarmField(BaseVarmField):
+    """
+    An AnnDataField for an .varm field in the AnnData data structure.
+    In addition to creating a reference to the .varm field, stores the column
+    keys for the varm field in a more accessible .uns attribute.
+    Parameters
```
Comment (martinkim0) [inline scvi/data/fields/_varm_field.py:47]:
```suggestion
    An AnnDataField for an .varm field in the AnnData data structure.
    
    In addition to creating a reference to the .varm field, stores the column
    keys for the varm field in a more accessible .uns attribute.
    
    Parameters
```

File: scvi/data/fields/_varm_field.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        if self.is_count_data and not _check_nonnegative_integers(varm_data):
+            warnings.warn(
+                f"adata.varm['{self.attr_key}'] does not contain unnormalized count data. "
+                "Are you sure this is what you want?"
+            )
+
+    def _setup_column_names(self, adata: AnnData) -> Union[list, np.ndarray]:
+        """
+        Returns a list or NumPy array of column names that will be used for the relevant .varm data.
+        If the ``colnames_uns_key`` was specified, then the columns stored in that
```
Comment (martinkim0) [inline scvi/data/fields/_varm_field.py:107]:
```suggestion
        Returns a list or NumPy array of column names that will be used for the relevant .varm data.
        
        If the ``colnames_uns_key`` was specified, then the columns stored in that
```

**Review by martinkim0 — APPROVED** [review]
LGTM, ready to merge after precommit and Codacy issues are addressed!

**Review by adamgayoso — CHANGES_REQUESTED** [review]
Thanks! Would you be able to add the classes to our docs too?

**Comment by adamgayoso** [comment]
Actually, thinking about this again, there's going to be a lot of copy/paste between the obs analogs. I think it makes sense to go deeper with the abstraction so we can reduce code duplication. We are on it from our side @vitkl

**Comment by martinkim0** [comment]
Closed in favor of #1821

**Pull request closed without merging**